### PR TITLE
python3Packages.mrsqm: init at 0.0.4

### PR DESCRIPTION
**Pull request by mbalatsko**
## Description of changes
Adding one of the requirements needed for https://github.com/NixOS/nixpkgs/issues/247089 . I want to separate `mindsdb` requirements initialization into separate PRs bacause adding it to `nixpkgs` requires more than 15 new packages. I believe this way it would be easier to review.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Result of `nixpkgs-review pr 255532` on x86_64-linux:
```
2 packages built:
python310Packages.mrsqm python311Packages.mrsqm
```

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.mrsqm python311Packages.mrsqm

File: pkgs/development/python-modules/mrsqm/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  format = "setuptools";
+
+  disable = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-kg9GSgtBpnCF+09jyP5TRwZh0tifxx4WRtQGn8bLH8c=";
+  };
+
+  buildInputs = [ fftw.dev ];
```
Comment (natsukium) [inline pkgs/development/python-modules/mrsqm/default.nix:24]:
```suggestion
  buildInputs = [ fftw ];
```

File: pkgs/development/python-modules/mrsqm/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    pandas
+    scikit-learn
+    numpy
+  ];
+
+  doCheck = false; # Package has no tests
+  pythonImportsCheck = [ "mrsqm" ];
+
+  meta = with lib; {
+    description = "MrSQM (Multiple Representations Sequence Miner) is a time series classifier.";
```
Comment (natsukium) [inline pkgs/development/python-modules/mrsqm/default.nix:40]:
```suggestion
    description = "MrSQM (Multiple Representations Sequence Miner) is a time series classifier";
```

File: pkgs/development/python-modules/mrsqm/default.nix (hunk 1 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/mrsqm/default.nix]: fixed

File: pkgs/development/python-modules/mrsqm/default.nix (hunk 2 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/mrsqm/default.nix]: fixed

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.mrsqm python311Packages.mrsqm

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.mrsqm python311Packages.mrsqm

**Comment by mbalatsko** [comment]
There was some unexpected error in previous build, reruning

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 255532` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.mrsqm</li>
    <li>python310Packages.mrsqm.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.mrsqm</li>
    <li>python311Packages.mrsqm.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 255532` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.mrsqm</li>
    <li>python310Packages.mrsqm.dist</li>
    <li>python311Packages.mrsqm</li>
    <li>python311Packages.mrsqm.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
Thanks!

**Pull request merged**